javascript/javascript1/week4/voiceAssistant.js

```diff
@@ -0,0 +1,183 @@
+        return setTimer(command)
+    }
+    
+    if (command.startsWith('remind me to')) {
+        return setReminder(command)
+    }
+
+    else{
+        return `Sorry, I don't understand your command`
+    }
+}
+
+
+function userNameLogin(command) {
+    let loginName = command.split(' ')[4].toLowerCase().trim();
+    if (loginName === '') {
+        return 'Please tell me your name'
+    }else if (loginName === userName) {
+        return `Nice to meet you again, ${userName}`
+    }else{
+        userName = loginName;
+        return `Nice to meet you ${userName}`
+    }
+}
+
+
+function replyMyName() {
+    if(userName != ''){
+        return `Your name is ${userName}`
+    }else return 'Please tell me your name'
+}
+
+
+function addToDoList(command) {
+    const addStartIndex = command.indexOf('add') + 4;
+    const addEndIndex = command.indexOf('to my todo');
+    let thingToDo = command.slice(addStartIndex, addEndIndex).trim();
+    if (thingToDo === '') {
+        return 'What do you want to add to your todo?'
+    }else if (toDoList.includes(thingToDo)) {
+        return `You have already added ${thingToDo} to your todo`
+    }else{
+        toDoList.push(thingToDo);
+        return `${thingToDo} added to your todo`
+    }
+}
+
+
+function removeToDoList(command) {
+    const removeStartIndex = command.indexOf('remove') + 7;
+    const removeEndIndex = command.indexOf('from my todo');
+    let thingToRemove = command.slice(removeStartIndex, removeEndIndex).trim();
+    if (thingToRemove === '') {
+        return 'What do you want to remove from your todo?'
+    }else if (!toDoList.includes(thingToRemove)) {
+        return `You don't have ${thingToRemove} on your todo list`
+    }else{
+        let IndexOfThingToRemove = toDoList.indexOf(thingToRemove);
+        toDoList.splice(IndexOfThingToRemove, 1)
+        return `Remove ${thingToRemove} from your todo`
+    }
+}
+
+
+function showToDoList() {
+    if (toDoList.length === 0) {
+        return 'There is no todo on the list'
+    }else if (toDoList.length === 1) {
+        return `You have ${toDoList.length} todo - ${toDoList[0]}.`
+    }else{
+        let tasksOutput = toDoList.slice(0, toDoList.length - 1).join(', ');
+        let lastTask = toDoList[toDoList.length - 1]
+        return `You have ${toDoList.length} todos - ${tasksOutput} and ${lastTask}.`
+    }
+}
+
+
+function findTodayDate() {
+    let todayDate = new Date();
+    let todayDateYear = todayDate.getFullYear();
+    const allMonths = ['January', 'Feburary', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December']
+    let todayDateMonth = allMonths[todayDate.getMonth()]
+    let todayDateDay= todayDate.getDate();
+
+    return `Today is ${todayDateDay}. of ${todayDateMonth} ${todayDateYear}` 
+}
+
+
+function numberCalculator(command) {
+    let calculatedString = command.slice(8).trim();
+    let result = eval(calculatedString);
+    return result;
```

Comment:
"eval"  is easy and working approach but it's generally not recommended to use due to potential code injection attacks as eval() executes any code passed to it